### ISSUE TITLE
UERANSIM: [nas] [error] Service Reject received with cause [UE_IDENTITY_CANNOT_BE_DERIVED_FROM_NETWORK]

### ISSUE THREAD

**Issue by pau-alef** [giros-dit/containerized-cellular-testbeds#2]
When launching the UE simulator in the `5g-sa_open5gs_ueransim` scenario I'm getting the following error, which seems to be preventing connectivity:

```
$ ./start-ue1.sh 
Starting UERANSIM UE 1 (in foreground)...


UERANSIM v3.2.6
[2024-01-08 10:27:59.659] [nas] [info] UE switches to state [MM-DEREGISTERED/PLMN-SEARCH]
[2024-01-08 10:27:59.660] [rrc] [debug] New signal detected for cell[1], total [1] cells in coverage
[2024-01-08 10:27:59.660] [nas] [info] Selected plmn[001/01]
[2024-01-08 10:27:59.660] [rrc] [info] Selected cell plmn[001/01] tac[1] category[SUITABLE]
[2024-01-08 10:27:59.660] [nas] [info] UE switches to state [MM-DEREGISTERED/PS]
[2024-01-08 10:27:59.660] [nas] [info] UE switches to state [MM-DEREGISTERED/NORMAL-SERVICE]
[2024-01-08 10:27:59.660] [nas] [debug] Initial registration required due to [MM-DEREG-NORMAL-SERVICE]
[2024-01-08 10:27:59.660] [nas] [debug] UAC access attempt is allowed for identity[0], category[MO_sig]
[2024-01-08 10:27:59.660] [nas] [debug] Sending Initial Registration
[2024-01-08 10:27:59.660] [nas] [info] UE switches to state [MM-REGISTER-INITIATED]
[2024-01-08 10:27:59.660] [rrc] [debug] Sending RRC Setup Request
[2024-01-08 10:27:59.660] [rrc] [info] RRC connection established
[2024-01-08 10:27:59.660] [rrc] [info] UE switches to state [RRC-CONNECTED]
[2024-01-08 10:27:59.660] [nas] [info] UE switches to state [CM-CONNECTED]
[2024-01-08 10:27:59.663] [nas] [debug] Authentication Request received
[2024-01-08 10:27:59.663] [nas] [debug] Received SQN [000000000000]
[2024-01-08 10:27:59.663] [nas] [debug] SQN-MS [000000000000]
[2024-01-08 10:27:59.663] [nas] [debug] Sending Authentication Failure due to SQN out of range
[2024-01-08 10:27:59.666] [nas] [debug] Authentication Request received
[2024-01-08 10:27:59.666] [nas] [debug] Received SQN [000000000021]
[2024-01-08 10:27:59.666] [nas] [debug] SQN-MS [000000000000]
[2024-01-08 10:27:59.668] [nas] [debug] Security Mode Command received
[2024-01-08 10:27:59.668] [nas] [debug] Selected integrity[2] ciphering[0]
[2024-01-08 10:27:59.674] [nas] [debug] Registration accept received
[2024-01-08 10:27:59.674] [nas] [info] UE switches to state [MM-REGISTERED/NORMAL-SERVICE]
[2024-01-08 10:27:59.674] [nas] [debug] Sending Registration Complete
[2024-01-08 10:27:59.674] [nas] [info] Initial Registration is successful
[2024-01-08 10:27:59.674] [nas] [debug] Sending PDU Session Establishment Request
[2024-01-08 10:27:59.674] [nas] [debug] UAC access attempt is allowed for identity[0], category[MO_sig]
[2024-01-08 10:27:59.876] [nas] [debug] Configuration Update Command received
[2024-01-08 10:27:59.891] [nas] [debug] PDU Session Establishment Accept received
[2024-01-08 10:27:59.892] [nas] [info] PDU Session establishment is successful PSI[1]
[2024-01-08 10:27:59.917] [app] [info] Connection setup for PDU session[1] is successful, TUN interface[uesimtun0, 172.45.1.2] is up.
[2024-01-08 10:32:47.762] [rrc] [debug] RRC Release received
[2024-01-08 10:32:47.762] [nas] [info] UE switches to state [CM-IDLE]
[2024-01-08 10:37:41.015] [nas] [debug] Uplink data status changed PSI[1] pending[true]
[2024-01-08 10:37:41.015] [nas] [debug] Service request required due to [IDLE-UPLINK-DATA-PENDING]
[2024-01-08 10:37:41.015] [nas] [debug] UAC access attempt is allowed for identity[0], category[MO_data]
[2024-01-08 10:37:41.015] [nas] [debug] Sending Service Request due to [IDLE-UPLINK-DATA-PENDING]
[2024-01-08 10:37:41.015] [nas] [info] UE switches to state [MM-SERVICE-REQUEST-INITIATED]
[2024-01-08 10:37:41.015] [rrc] [debug] Sending RRC Setup Request
[2024-01-08 10:37:41.017] [rrc] [info] RRC connection established
[2024-01-08 10:37:41.017] [rrc] [info] UE switches to state [RRC-CONNECTED]
[2024-01-08 10:37:41.017] [nas] [info] UE switches to state [CM-CONNECTED]
[2024-01-08 10:37:41.019] [rrc] [debug] RRC Release received
[2024-01-08 10:37:41.019] [nas] [warning] Not protected Service Reject message received
[2024-01-08 10:37:41.019] [nas] [error] Service Reject received with cause [UE_IDENTITY_CANNOT_BE_DERIVED_FROM_NETWORK]
[2024-01-08 10:37:41.019] [nas] [info] UE switches to state [5U2-NOT-UPDATED]
[2024-01-08 10:37:41.019] [nas] [info] UE switches to state [MM-DEREGISTERED/PS]
[2024-01-08 10:37:41.019] [nas] [debug] Initial registration required due to [DUE-TO-SERVICE_REJECT]
[2024-01-08 10:37:41.019] [nas] [info] UE switches to state [CM-IDLE]
[2024-01-08 10:37:41.019] [nas] [info] UE switches to state [MM-DEREGISTERED/NORMAL-SERVICE]
[2024-01-08 10:37:41.019] [nas] [debug] UAC access attempt is allowed for identity[0], category[MO_data]
[2024-01-08 10:37:41.019] [nas] [debug] Sending Initial Registration
[2024-01-08 10:37:41.019] [nas] [info] UE switches to state [MM-REGISTER-INITIATED]
[2024-01-08 10:37:41.019] [rrc] [debug] Sending RRC Setup Request
[2024-01-08 10:37:41.020] [rrc] [info] RRC connection established
[2024-01-08 10:37:41.020] [rrc] [info] UE switches to state [RRC-CONNECTED]
[2024-01-08 10:37:41.020] [nas] [info] UE switches to state [CM-CONNECTED]
[2024-01-08 10:37:41.234] [nas] [debug] Authentication Request received
[2024-01-08 10:37:41.234] [nas] [debug] Received SQN [000000000081]
[2024-01-08 10:37:41.234] [nas] [debug] SQN-MS [000000000021]
[2024-01-08 10:37:41.236] [nas] [debug] Security Mode Command received
[2024-01-08 10:37:41.237] [nas] [debug] Selected integrity[2] ciphering[0]
[2024-01-08 10:37:41.242] [nas] [debug] Registration accept received
[2024-01-08 10:37:41.242] [nas] [info] UE switches to state [MM-REGISTERED/NORMAL-SERVICE]
[2024-01-08 10:37:41.242] [nas] [info] UE switches to state [5U1-UPDATED]
[2024-01-08 10:37:41.242] [nas] [debug] Sending Registration Complete
[2024-01-08 10:37:41.242] [nas] [info] Initial Registration is successful
[2024-01-08 10:37:41.443] [nas] [debug] Configuration Update Command received
[2024-01-08 10:37:57.608] [rrc] [debug] RRC Release received
[2024-01-08 10:37:57.608] [nas] [info] UE switches to state [CM-IDLE]
```

**Comment by david-martinez-garcia** [comment]
Hello @pau-alef,

I can’t recall now if I have faced this particular issue before. However, I do remember that sometimes the first execution of the UE script failed. What if you stop the execution and try to run it again immediately? Does it run successfully? I am not able to test this right now.

Thanks.

**Comment by pau-alef** [comment]
> Hello @pau-alef,
> 
> I can’t recall now if I have faced this particular issue before. However, I do remember that sometimes the first execution of the UE script failed. What if you stop the execution and try to run it again immediately? Does it run successfully? I am not able to test this right now.
> 
> Thanks.

I think this happens when I spawn a 2nd UE, basically because the scripts use the same config for both UE, with overlapping IDs:

**Comment by david-martinez-garcia** [comment]
> > Hello @pau-alef,
> > I can’t recall now if I have faced this particular issue before. However, I do remember that sometimes the first execution of the UE script failed. What if you stop the execution and try to run it again immediately? Does it run successfully? I am not able to test this right now.
> > Thanks.
> 
> I think this happens when I spawn a 2nd UE, basically because the scripts use the same config for both UE, with overlapping IDs:

Thanks for the hint. As of now both UEs have the same IMSI. I’ll create another config file for UE 2 and upload the changes in the same commit that hopefully will also fix #1.
